src/daft-core/src/array/ops/binary.rs

```diff
@@ -0,0 +1,40 @@
+use common_error::DaftResult;
+
+use crate::{
+    array::ops::as_arrow::AsArrow,
+    datatypes::{BinaryArray, UInt64Array},
+};
+
+impl BinaryArray {
+    pub fn length(&self) -> DaftResult<UInt64Array> {
+        let self_arrow = self.as_arrow();
+        let arrow_result = self_arrow
+            .iter()
+            .map(|val| {
+                let v = val?;
+                Some(v.len() as u64)
+            })
+            .collect::<arrow2::array::UInt64Array>()
```

Comment:
This will work but a more efficient way may be to use the offset buffer directly instead of iterating through every value. https://docs.rs/arrow2/latest/arrow2/array/struct.BinaryArray.html#method.offsets